### PR TITLE
[86by4t6c9][switch] some a11y improvements

### PR DESCRIPTION
**Pull request by ilyabrower**
## Motivation and Context
Read more in changelog

## How has this been tested?
Manually

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [X] Nice improve.

## Checklist:

- [X] My code follows the code style of this project.
- [X] I have updated the documentation accordingly or it's not required.
- [X] Unit tests are not broken.
- [X] I have added changelog note to corresponding `CHANGELOG.md` file with planned publish date.
- [ ] I have added new unit tests on added of fixed functionality.

### CODE REVIEW

**Review by j-mnizhek — APPROVED** [review]
Looks good! ✨ I've pushed two commits with updates for documentation (from https://github.com/semrush/intergalactic/pull/1402) and examples for Switch.

File: semcore/switch/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,18 @@
 
 CHANGELOG.md standards are inspired by [keepachangelog.com](https://keepachangelog.com/en/1.0.0/).
 
+## [5.28.0] - 2024-06-11
+
+### Changed
+
+- set as deprecated `hidden` property on Switch.Addon.
```
Comment (msereniti) [inline semcore/switch/CHANGELOG.md:9]: Maybe we should better explain why we did it here?

File: semcore/switch/src/index.d.ts [hunk 2]
```diff
@@ -43,7 +43,14 @@ export type SwitchValueProps = BoxProps &
 
 /** @deprecated */
 export interface ISwitchAddonProps extends SwitchAddonProps, UnknownProperties {}
-export type SwitchAddonProps = BoxProps & NeighborItemProps & {};
+export type SwitchAddonProps = BoxProps &
+  NeighborItemProps & {
+    /** @deprecated
```
Comment (msereniti) [inline semcore/switch/src/index.d.ts:48]: Also this JSDoc tag is not parsing correctly
```
    /**
     * @deprecated
 ```
 
 Will work

Comment (ilyabrower) [reply semcore/switch/src/index.d.ts:48]: both options are correct

File: semcore/switch/CHANGELOG.md (hunk 1 above)
Comment (ilyabrower) [reply semcore/switch/CHANGELOG.md]: fixed

**Review by msereniti — APPROVED** [review]

**Pull request merged**